pallets/xtransfer/src/lib.rs

```diff
@@ -164,13 +173,164 @@ pub mod pallet {
 	pub enum Error<T> {
 		/// Bad location.
 		BadLocation,
+		/// Cannot send message from parachain to self
+		CannotSendToSelf,
+		/// Call to SendXcm to initiate open channel failed
+		FailedToSendChannelOpenRequest,
+		/// Maximum one channel per relation ~ (sender,receiver) and direction matters
+		MaxOneChannelPerRelation,
+		/// Requires existing open channel with self as sender
+		NoSenderChannelOpen,
+		/// Requires existing open channel with self as recipient
+		NoRecipientChannelOpen,
 	}
 
+	#[pallet::storage]
+	#[pallet::getter(fn sender_channels)]
+	/// Stores all parachains with which this parachain has opened a channel as the sender
+	/// -> may be redundant, but it propagates errors earlier than in processing by runtime relay
+	pub type SenderChannels<T: Config> = StorageValue<_, Vec<ParaId>, ValueQuery>;
```

Comment:
how are these updated when the other side closes it?

likewise for the RecipientChannels

Reply:
by DownwardMessageHandler impl